FhirToDataLake/src/Microsoft.Health.AnalyticsConnector.Core/Jobs/FhirToDataLakeProcessingJobSpliter.cs

```diff
@@ -0,0 +1,319 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Extensions.Logging;
+using Microsoft.Health.AnalyticsConnector.Common.Logging;
+using Microsoft.Health.AnalyticsConnector.Core.Exceptions;
+using Microsoft.Health.AnalyticsConnector.Core.Extensions;
+using Microsoft.Health.AnalyticsConnector.Core.Fhir;
+using Microsoft.Health.AnalyticsConnector.Core.Jobs.Models;
+using Microsoft.Health.AnalyticsConnector.DataClient;
+using Microsoft.Health.AnalyticsConnector.DataClient.Api.Fhir;
+using Microsoft.Health.AnalyticsConnector.DataClient.Extensions;
+using Microsoft.Health.AnalyticsConnector.DataClient.Models.FhirApiOption;
+using Microsoft.Health.JobManagement;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Polly.Timeout;
+
+namespace Microsoft.Health.AnalyticsConnector.Core.Jobs
+{
+    public class FhirToDataLakeProcessingJobSpliter
+    {
+        private readonly IApiDataClient _dataClient;
+        private readonly IDiagnosticLogger _diagnosticLogger;
+        private readonly ILogger<FhirToDataLakeProcessingJobSpliter> _logger;
+        private const string CountApiParameter = "count";
+        private const string TotalCountFieldKey = "total";
+        private const string LastUpdatedApiParameter = "_lastUpdated";
+        private const string LastUpdatedApiParameterDesc = "-_lastUpdated";
+        private const int TimeoutSecondsGetResourceCount = 90;
```

Comment:
I feel this timeout value can be aligned with FhirApiClient

Reply:
It's too long for FHIR server throw time out exception (around 3 mins), maybe we can early cancel the operation.